roboteq_diff_driver/src/driver.cpp

```diff
@@ -1,198 +1,4 @@
-#endif
-
-
-
-void mySigintHandler(int sig)
-{
-  ROS_INFO("Received SIGINT signal, shutting down...");
-  ros::shutdown();
-}
-
-
-uint32_t millis()
-{
-	ros::WallTime walltime = ros::WallTime::now();
-//	return (uint32_t)((walltime._sec*1000 + walltime.nsec/1000000.0) + 0.5);
-//	return (uint32_t)(walltime.toNSec()/1000000.0+0.5);
-	return (uint32_t)(walltime.toNSec()/1000000);
-}
-
-class MainNode
-{
-
-public:
-  MainNode();
-
-public:
-
-  //
-  // cmd_vel subscriber
-  //
-  void cmdvel_callback( const geometry_msgs::Twist& twist_msg);
-  void cmdvel_setup();
-  void cmdvel_loop();
-  void cmdvel_run();
-
-  //
-  // odom publisher
-  //
-  void odom_setup();
-  void odom_stream();
-  void odom_loop();
-  //void odom_hs_run();
-  void odom_ms_run();
-  void odom_ls_run();
-  void odom_publish();
-#ifdef _ODOM_COVAR_SERVER
-  void odom_covar_callback(const roboteq_diff_msgs::RequestOdometryCovariancesRequest& req, roboteq_diff_msgs::RequestOdometryCovariancesResponse& res);
-#endif
-
-  int run();
-
-protected:
-  ros::NodeHandle nh;
-  ros::NodeHandle nhLocal;
-
-  serial::Serial controller;
-
-  uint32_t starttime;
-  uint32_t hstimer;
-  uint32_t mstimer;
-  uint32_t lstimer;
-
-  //
-  // cmd_vel subscriber
-  //
-  ros::Subscriber cmdvel_sub;
-
-  //
-  // odom publisher
-  //
-  geometry_msgs::TransformStamped tf_msg;
-  tf::TransformBroadcaster odom_broadcaster;
-  nav_msgs::Odometry odom_msg;
-  ros::Publisher odom_pub;
-
-  roboteq_diff_msgs::WheelRates wheel_rates_msg;
-  roboteq_diff_msgs::WheelRates wheel_rates_desired_msg;
-  ros::Publisher wheel_rate_pub;
-  ros::Publisher wheel_rate_desired_pub;
-
-#ifdef _ODOM_SENSORS
-  std_msgs::Float32 voltage_msg;
-  ros::Publisher voltage_pub;
-  roboteq_diff_msgs::Duplex current_msg;
-  ros::Publisher current_pub;
-  std_msgs::Float32 energy_msg;
-  ros::Publisher energy_pub;
-  std_msgs::Float32 temperature_msg;
-  ros::Publisher temperature_pub;
-#endif
-
-  // buffer for reading encoder counts
-  int odom_idx;
-  char odom_buf[24];
-
-  // toss out initial encoder readings
-  char odom_encoder_toss;
-
-  int32_t odom_encoder_left;
-  int32_t odom_encoder_right;
-
-  float odom_x;
-  float odom_y;
-  float odom_yaw;
-  float odom_last_x;
-  float odom_last_y;
-  float odom_last_yaw;
-
-  uint32_t odom_last_time;
-
-#ifdef _ODOM_SENSORS
-  float voltage;
-  float current_right;
-  float current_left;
-  float energy;
-  float temperature;
-  uint32_t current_last_time;
-#endif
-
-  // settings
-  bool pub_odom_tf;
-  std::string odom_frame;
-  std::string base_frame;
-  std::string cmdvel_topic;
-  std::string odom_topic;
-  std::string port;
-  int baud;
-  bool open_loop;
-  double wheel_circumference;
-  double track_width;
-  int encoder_ppr;
-  int encoder_cpr;
-  double max_amps;
-  int max_rpm;
-  double roboteq_rpm_scale;
-  double ros_loop_rate;
-  double serial_timeout_ms;
-
-};
+#include "../include/roboteq_diff_driver/driver.h"
```

Comment:
If the CMakelists is set up right, you should be able to do `#include "roboteq_diff_driver/driver.h"` If you have trouble with getting that to work, let me know

Reply:
Fixed